cmd/rpcdaemon/commands/trace_filtering.go

```diff
@@ -899,14 +901,16 @@ func (api *TraceAPIImpl) filterV3(ctx context.Context, dbtx kv.TemporalTx, fromB
 	return stream.Flush()
 }
 
-func filter_trace(pt *ParityTrace, fromAddresses map[common.Address]struct{}, toAddresses map[common.Address]struct{}) bool {
+func filter_trace(pt *ParityTrace, fromAddresses map[common.Address]struct{}, toAddresses map[common.Address]struct{}, isIntersectionMode bool) bool {
 	switch action := pt.Action.(type) {
 	case *CallTraceAction:
 		_, f := fromAddresses[action.From]
 		_, t := toAddresses[action.To]
 
-		if f || t {
-			return true
+		if isIntersectionMode {
+			return f && t
+		} else {
+			return f || t
 		}
 	case *CreateTraceAction:
 		_, f := fromAddresses[action.From]
```

Comment:
this case also need handle?

Reply:
okay, I handled this case, and set t as false if res.Address is nil

Reply:
Perhaps it would be more user-friendly to match the "0x000000..." toAddress when res.Address is nil. The caller can specify the 0x00000... address in toAddresses to filter transactions created by contracts. @AskAlexSharov

Reply:
I'm sorry, please ignore the previous message. The address used for lookup during contract creation is not action.To.